src/test/unit_tests/test_lifecycle.js

```diff
@@ -583,7 +631,142 @@ mocha.describe('lifecycle', () => {
+            // remove all the noncurrent versions expect one( newnon_current_version = 1)  once 15 days passed 
+            // current version has not expired. In that case, list will return 2 objects
+            await verify_version_deleted(2, version_bucket_key);
+        });
+
+        mocha.it('lifecyle - version expiration - only NewerNoncurrentVersions exceeded', async () => {
+            const days = 35;
+            const version_count = 10;
+            const newnon_current_version = 5;
+            const noncurrent_days = 30;
+            const version_bucket_key = 'test-lifecycle-version1-1';
+            // create_time updated to 29 days and expire is 30 days, do not delete any multipart
+            await create_mock_version(version_bucket_key, version_bucket, days + 1, version_count, true);
+            const putLifecycleParams = commonTests.version_lifecycle_configuration(version_bucket,
+                                            version_bucket_key, days, newnon_current_version, noncurrent_days);
+
+            await s3.putBucketLifecycleConfiguration(putLifecycleParams);
+            await lifecycle.background_worker();
+            // object will expire, noncurrent version count(5) + deleted current version(1) + delete marker(1)
+            await verify_version_deleted(7, version_bucket_key);
+        });
+
+        mocha.it('lifecyle - version expiration - only NoncurrentDays exceeded', async () => {
+            const days = 45;
+            const version_count = 10;
+            const newnon_current_version = 100;
+            const noncurrent_days = 30;
+            const version_bucket_key = 'test-lifecycle-version3';
+            // create_time updated to 29 days and expire is 30 days, do not delete any multipart
+            await create_mock_version(version_bucket_key, version_bucket, days + noncurrent_days + 1, version_count, true);
+            const putLifecycleParams = commonTests.version_lifecycle_configuration(version_bucket,
+                                            version_bucket_key, days, newnon_current_version, noncurrent_days);
+
+            await s3.putBucketLifecycleConfiguration(putLifecycleParams);
+            await lifecycle.background_worker();
+            // noncurrent version count + delete marker(1)
+            await verify_version_deleted(11, version_bucket_key);
+        });
+
+        mocha.it('lifecyle - version expiration - both NoncurrentDays and NewerNoncurrentVersions exceeded', async () => {
+            const days = 30;
+            const version_count = 10;
+            //const expired_delete_marker = false;
+            const newnon_current_version = 1;
+            const noncurrent_days = 15;
+            const version_bucket_key = 'test-lifecycle-version4';
+            // create_time updated to 29 days and expire is 30 days, do not delete any multipart
+            await create_mock_version(version_bucket_key, version_bucket, days + noncurrent_days + 1, version_count);
```

Comment:
why expire_all not true here?

Reply:
we are not expiring current version, create_time updated only for the noncurret version

Reply:
no delete marker test here, `version_lifecycle_configuration` only have version conditions, but at the end its works as delete marker false

Reply:
got it, Thanks